src/NuGet.Core/NuGet.Frameworks/CompatibilityProvider.cs

```diff
@@ -123,13 +116,16 @@ private bool IsPCLCompatible(NuGetFramework target, NuGetFramework candidate)
                 return IsCompatibleWithTarget(target, candidate);
             }
 
-            IEnumerable<NuGetFramework> targetFrameworks;
-            IEnumerable<NuGetFramework> candidateFrameworks;
+            IEnumerable<NuGetFramework>? targetFrameworks;
+            IEnumerable<NuGetFramework>? candidateFrameworks;
 
             if (target.IsPCL)
             {
                 // do not include optional frameworks here since we might be unable to tell what is optional on the other framework
-                _mappings.TryGetPortableFrameworks(target.Profile, false, out targetFrameworks);
+                if (!_mappings.TryGetPortableFrameworks(target.Profile, false, out targetFrameworks))
```

Comment:
Since you're here, do you want to add parameter names to constants?
```suggestion
                if (!_mappings.TryGetPortableFrameworks(target.Profile, includeOptional: false, out targetFrameworks))
```